WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift

```diff
@@ -48,7 +50,28 @@ public class ProductsScreen: ScreenObject {
 
     @discardableResult
     public func selectProduct(atIndex index: Int) throws -> SingleProductScreen {
-        XCUIApplication().tables.cells.element(boundBy: index).tap()
+        app.tables.cells.element(boundBy: index).tap()
+        return try SingleProductScreen()
+    }
+
+    public func selectProduct(byName name: String) throws -> SingleProductScreen {
+        app.tables.cells.staticTexts[name].tap()
         return try SingleProductScreen()
     }
+
+    public func verifyProductListOnProductsScreen(count: Int, name: String, status: String) throws -> Self {
+        XCTAssertTrue(try app.getTextVisibilityCount(text: name) == 1, "Product name does not exist!")
+        XCTAssertTrue(app.tables.cells.count == count, "Expecting \(count) products, got \(app.tables.cells.count) instead!")
+
+        /// TODO: Add AccessibilityIdentifiers to product list cell elements to be able to get stock status label value and update this assert.
+        /// Currently mock response API has "in stock" which is set to the first product repeated 3 times, so current assert will check that it appears 3 times
```

Comment:
There's a difference between a `//` and `///` comment. The slashes are the documentation comments, and are often used at the top of a function to describe it. They support markdown and I assume handled differently than usual comments by Swift. How this works: hitting `Option`+`Click` on a function call will show a function description.

In this case this is just a note (simple comment), so I believe `//` is more appropriate here:

```suggestion
        // TODO: Add AccessibilityIdentifiers to product list cell elements to be able to get stock status label value and update this assert.
        // Currently mock response API has "in stock" which is set to the first product repeated 3 times, so current assert will check that it appears 3 times
```